src/qtism/runtime/pci/json/Unmarshaller.php

```diff
@@ -380,21 +385,55 @@ protected function unmarshallDuration(array $unit)
     }
 
     /**
-     * Unmarshall a duration JSON PCI representation.
+     * Unmarshall an uploaded file payload JSON PCI representation.
      *
      * @param array $unit
      * @return QtiFile
      * @throws FileManagerException
      */
     protected function unmarshallFile(array $unit)
     {
-        $filename = (empty($unit['base']['file']['name']) === true) ? '' : $unit['base']['file']['name'];
+        $fileArray = $unit['base']['file'];
+        return $this->getFileManager()->createFromData(
+            base64_decode($fileArray['data']),
+            $fileArray['mime'],
+            $fileArray['name'] ?? ''
+        );
+    }
+
+    /**
+     * Unmarshall an uploaded file hash JSON PCI representation.
+     * 
+     * This is not a standard QTI feature but a convenience to store only
+     * a hash of the file, to avoid storing huge files in the test session.
+     * This suppose the following:
+     * * Payload of the file has been previously persisted before.
+     * * "name" key contains the persisted file path with proper wrapper. 
+     * * "data" key contains the hash, not base64_encode'd.
```

Comment:
I see you use `base64_encode` in `marshallFile` method

Reply:
Yes, I think it is more consistent to keep the same logic with FileSystemFile when marshalling / unmarshalling.